### PR TITLE
Adapting CLI to parse consensus parameters

### LINKED ISSUE #894: Turn necessary consensus-critical constant into argument
At the moment, we have issues with consensus constants being constant and accessed through the code source. E.g. we have the various genesis values, but also the cycle length.

The problem with such constants is that they might need to be modified to ensure things work correctly at a low cost. E.g. our cycles are 1008 blocks at the moment, which is expensive to generate in a test. Being able to reduce this to a few blocks would make things more scalable.

There's three approaches I can think of:
- Having them with `version` => This is what we're doing currently. It works well when one need not to change those values, or when there's a "common default". E.g. we have a common genesis block for all unittests. However it creates problems for integration testing as we need a special build.

- Having them global with selected with a runtime parameter. This is what we started to do with the genesis block, e.g. #712 , but it also creates some issues, e.g. #813. This issue recently showed again with `ValidatorCycle`.

- Having a global context that is passed around. This would be similar to the idea of "Consensus params" in the BTC implementation: https://github.com/bitcoin/bitcoin/blob/master/src/consensus/params.h but might lead to a lot of context being passed around if overused.

Note that BTC does have a mixture of both 2 and 3: https://github.com/bitcoin/bitcoin/blob/master/src/consensus/consensus.h

### PR DESCRIPTION
**Pull request by linked0**
After making the consensus-critical constants to be propagated through parameters, we adapt the CLI to parse the constants.

Fixes #894

### CODE REVIEW

File: source/agora/cli/multi/main.d [hunk 1]
```diff
@@ -106,7 +106,11 @@ private int main (string[] args)
 
     FullNode[] nodes;
     foreach (const ref config; configs)
-        nodes ~= runNode(config, new immutable(ConsensusParams)());
+    {
+        auto params = new immutable(ConsensusParams)(
+            config.node.validator_cycle);
+        nodes ~= runNode(config, params);
```
Comment (Geod24) [inline source/agora/cli/multi/main.d:112]: Why the separation though ?

**Comment by Geod24** [comment]
I have a 99.99% certitude this doesn't work.

File: source/agora/cli/multi/main.d (hunk 1 above)
Comment (linked0) [reply source/agora/cli/multi/main.d:112]: Not much meaning. It's just for more readable code.

**Comment by linked0** [comment]
> I have a 99.99% certitude this doesn't work.

I found the missing part. Sorry for that.

**Comment by linked0** [comment]
Ready for review.

File: source/agora/common/Config.d [hunk 2]
```diff
@@ -291,6 +294,7 @@ private NodeConfig parseNodeConfig (Node* node, const ref CommandLine cmdln)
 
     string data_dir = get!(string, "node", "data_dir")(cmdln, node);
     auto port = get!(ushort, "node", "port")(cmdln, node);
+    auto validator_cycle = get!(uint, "node", "validator_cycle")(cmdln, node);
```
Comment (Geod24) [inline source/agora/common/Config.d:297]: This needs to be optional

Comment (linked0) [reply source/agora/common/Config.d:297]: > This needs to be optional

How about this?
```
auto value = get!(uint, "node", "validator_cycle")(cmdln, node);
auto validator_cycle = (value != 0) ? value : NodeConfig.init.validator_cycle;
```

Comment (Geod24) [reply source/agora/common/Config.d:297]: There is `opt` for this

Comment (Geod24) [reply source/agora/common/Config.d:297]: Ah, my bad, I just realized that `get` also uses the default value.

**Comment by Geod24** [comment]
Reverted to a previous version (my feedback was wrong), will merge when green.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**